### ISSUE TITLE
http_parser.h:32:10: fatal error: 'sys/types.h' file not found

### ISSUE THREAD

**Issue by SimonDanisch** [JuliaWeb/HttpParser.jl#65]
This happens when executing `Pkg.build("Interact", "HttpParser")` on Julia 0.5, mid 2012 Mac running 10.11.5:
```
INFO: Building Interact

INFO: Enabling widgetsnbextension

Enabling notebook extension jupyter-js-widgets/extension...

      - Validating: OK

INFO: ipywidgets version found: 5.2.2

INFO: A compatible version of ipywidgets was found. All good.

INFO: Building HttpParser

INFO: Attempting to Create directory /Users/Boryana/.julia/v0.5/HttpParser/deps/downloads

INFO: Directory /Users/Boryana/.julia/v0.5/HttpParser/deps/downloads already created

INFO: Attempting to Create directory /Users/Boryana/.julia/v0.5/HttpParser/deps/downloads

INFO: Directory /Users/Boryana/.julia/v0.5/HttpParser/deps/downloads already created

INFO: Downloading file https://github.com/nodejs/http-parser/archive/v2.7.1.zip

INFO: Done downloading file https://github.com/nodejs/http-parser/archive/v2.7.1.zip

INFO: Attempting to Create directory /Users/Boryana/.julia/v0.5/HttpParser/deps

INFO: Directory /Users/Boryana/.julia/v0.5/HttpParser/deps already created

INFO: Path /Users/Boryana/.julia/v0.5/HttpParser/deps/src/http-parser-2.7.1 already created

INFO: Attempting to Create directory /Users/Boryana/.julia/v0.5/HttpParser/deps/usr/lib

INFO: Directory /Users/Boryana/.julia/v0.5/HttpParser/deps/usr/lib already created

INFO: Changing Directory to /Users/Boryana/.julia/v0.5/HttpParser/deps/src/http-parser-2.7.1

INFO: Changing Directory to /Users/Boryana/.julia/v0.5/HttpParser/deps/src/http-parser-2.7.1

INFO: Changing Directory to /Users/Boryana/.julia/v0.5/HttpParser/deps/src

INFO: Attempting to Create directory /Users/Boryana/.julia/v0.5/HttpParser/deps/usr/lib

INFO: Directory /Users/Boryana/.julia/v0.5/HttpParser/deps/usr/lib already created

INFO: Changing Directory to /Users/Boryana/.julia/v0.5/HttpParser/deps/src

cc  -I. -DHTTP_PARSER_STRICT=0  -Wall -Wextra -Werror -O3  -fPIC -c http_parser.c -o libhttp_parser.o

In file included from http_parser.c:24:

./http_parser.h:32:10: fatal error: 'sys/types.h' file not found

#include <sys/types.h>

         ^

1 error generated.

make: *** [libhttp_parser.o] Error 1

==================================================================[ ERROR: HttpParser ]===================================================================

 

LoadError: failed process: Process(setenv(`make -j8 -C http-parser-2.7.1 library`,String["XPC_FLAGS=0x0","PATH=/Applications/Julia-0.5.app/Contents/Resources/julia/bin:/Users/Boryana/.julia/v0.5/Homebrew/deps/usr/bin:/Users/Boryana/.julia/v0.5/Homebrew/deps/usr/sbin:/Applications/Julia-0.5.app/Contents/Resources/julia/bin:/Applications/anaconda/bin:/Library/Frameworks/Python.framework/Versions/Current/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/opt/X11/bin","PWD=/Users/Boryana","DISPLAY=/private/tmp/com.apple.launchd.YJHzuuoZT9/org.macosforge.xquartz:0","XPC_SERVICE_NAME=0","TERM_PROGRAM=Apple_Terminal","SHELL=/bin/bash","__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0","SONAME=libhttp_parser.dylib","TMPDIR=/var/folders/fp/djgc9ypn0_9c9d0myll6p3ym0000gn/T/","TK_LIBRARY=/System/Library/Frameworks/Tk.framework/Versions/8.5/Resources/Scripts","HOMEBREW_CACHE=/Users/Boryana/Library/Caches/Homebrew.jl/","FONTCONFIG_PATH=/Applications/Julia-0.5.app/Contents/Resources/julia/etc/fonts","LANG=en_US.UTF-8","SHLVL=0","LOGNAME=Boryana","TERM_SESSION_ID=09EA8BA6-8B60-4FE3-896F-9D1C9106AB8B","SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.cBtKSYL3PY/Listeners","Apple_PubSub_Socket_Render=/private/tmp/com.apple.launchd.w4bBjhP8KH/Render","USER=Boryana","SECURITYSESSIONID=186a7","HOMEBREW_NO_AUTO_UPDATE=1","TERM=xterm-256color","HOME=/Users/Boryana","TERM_PROGRAM_VERSION=361.1","OPENBLAS_MAIN_FREE=1","HOMEBREW_NO_ANALYTICS=1"]), ProcessExited(2)) [2]

while loading /Users/Boryana/.julia/v0.5/HttpParser/deps/build.jl, in expression starting on line 71

 

==========================================================================================================================================================

 

=====================================================================[ BUILD ERRORS ]=====================================================================

 

WARNING: HttpParser had build errors.

 

 - packages with build errors remain installed in /Users/Boryana/.julia/v0.5

 - build the package(s) and all dependencies with `Pkg.build("HttpParser")`

 - build a single package by running its `deps/build.jl` script
```

Are these missing dependencies?

Best,
Simon

**Comment by arnim** [comment]
Hi, same here @SimonDanisch have you found a solution?

**Comment by SimonDanisch** [comment]
Maybe!
You can try to install the Xcode library for apple (from http://railsapps.github.io/xcode-command-line-tools.html) and build again!
Let me know if that works!

**Comment by arnim** [comment]
Thank you @SimonDanisch ,

on Ubuntu 16.04 `sudo apt-get install libc6-dev-i386` helped  (by @rakanua)

**Comment by bnoordhuis** [comment]
FWIW, this should be fixed in http_parser>=2.8.0, it no longer depends on sys/types.h

Ref: https://github.com/nodejs/http-parser/pull/360
